src/core_plugins/metrics/server/lib/vis_data/helpers/get_splits.js

```diff
@@ -18,6 +19,14 @@ export default function getSplits(resp, panel, series) {
         bucket.label = formatKey(bucket.key, series);
         bucket.color = panel.type === 'top_n' ? color.hex() : colors.shift();
         bucket.meta = meta;
+        if (metricTypes.includes(panel.type) && panel.timerange_mode === 'all') {
```

Comment:
The `isMetric` helper could be used here and further below in this file as well.